### PR TITLE
[STY] Improve object formatting in the glossary

### PR DESCRIPTION
**Pull request by tsalo**
Closes None. I'm hoping that these changes will make the glossary more useful, since we'll be linking to it more and more as time goes by.

Changes proposed:

- Include object type-specific fields in glossary entries, such as "Allowed values" for enums.
- Bold categories in entries.
- Add format field.
- Prettify the schema information.

### CODE REVIEW

File: tools/schemacode/bidsschematools/render.py [hunk 1]
```diff
@@ -149,14 +170,16 @@ def make_glossary(schema, src_path=None):
             new_name = f"{new_name} ({group})"
             all_objects[new_name] = {}
             all_objects[new_name]["key"] = f"objects.{group}.{key}"
+            all_objects[new_name]["type"] = TYPE_CONVERTER.get(group, group)
```
Comment (bendhouseart) [inline tools/schemacode/bidsschematools/render.py:173]: cheeky 👍

**Review by bendhouseart — APPROVED** [review]

**Review by Remi-Gau — APPROVED** [review]
Looks good to me but we should probably link to the schema README if people want more info on how to understand the extra info, no?

BTW: we should probably update that schema README, no? It feels like the format info could be refactored into its own section

**Comment by tsalo** [comment]
I agree that we should link to the README more, but I'm not sure how to do that. I guess I could link to the `master` schema README, but I think it would be better to link to the appropriate commit's version.

I'm sure we'll update the README at the end of the sprint or some time next week.

**Comment by tsalo** [comment]
@Remi-Gau WDYT about pushing that off until we have #1154 dealt with? That way, we can just link to another page, instead of trying to hunt down the appropriate version on GitHub?

**Comment by Remi-Gau** [comment]
Totally fine with me.

**Comment by tsalo** [comment]
Cool! I'll merge this now then.

**Pull request merged**